tests/test_flow/test_persistence_random.py

```diff
@@ -108,15 +110,28 @@ def _add_binding_to_flow(self, binding):
 
         joined_dep_entity_names = ", ".join(binding.dep_entity_names)
 
+        # TODO Could this logic live inside ModelBinding instead?
```

Comment:
I was going to suggest this for `compute_func` at least. But I see why you have it here now because of the output decorator.

Reply:
Yes, I think this sentiment applies to both of these functions. Unfortunately they both need some global context (mainly `__descriptors_computed_by_XXX`), but that could be factored out.